internal/clients/iampolicy/iam_binding_batcher.go

```diff
@@ -184,45 +186,68 @@ func (f *policyFuture) executeModifers(ctx context.Context, u ResourceIamUpdater
 		return
 	}
 
-	// Get the existing policy
-	ep, diags := u.GetResourceIamPolicy(ctx)
-	if diags.HasError() {
-		f.set(nil, diags)
-		return
-	}
+	backoff := time.Second
+	maxBackoff := 30 * time.Second
 
-	// Determine the principal's we need to lookup their type.
-	principalSet, diags := f.getPrincipals(ctx, client)
-	if diags.HasError() {
-		f.set(nil, diags)
-		return
-	}
+	for {
+		// Get the existing policy
+		ep, diags := u.GetResourceIamPolicy(ctx)
+		if diags.HasError() {
+			f.set(nil, diags)
+			return
+		}
 
-	// Remove any bindings needed
-	bindings := ToMap(ep)
-	for _, rm := range f.removers {
-		if members, ok := bindings[rm.RoleID]; ok {
-			delete(members, rm.Members[0].MemberID)
-			if len(members) == 0 {
-				delete(bindings, rm.RoleID)
-			}
+		// Determine the principal's we need to lookup their type.
+		principalSet, diags := f.getPrincipals(ctx, client)
+		if diags.HasError() {
+			f.set(nil, diags)
+			return
 		}
-	}
 
-	// Go through the setters and apply them
-	for _, s := range f.setters {
-		members, ok := bindings[s.RoleID]
-		if !ok {
-			members = make(map[string]*models.HashicorpCloudResourcemanagerPolicyBindingMemberType, 4)
-			bindings[s.RoleID] = members
+		// Remove any bindings needed
+		bindings := ToMap(ep)
+		for _, rm := range f.removers {
+			if members, ok := bindings[rm.RoleID]; ok {
+				delete(members, rm.Members[0].MemberID)
+				if len(members) == 0 {
+					delete(bindings, rm.RoleID)
+				}
+			}
 		}
 
-		members[s.Members[0].MemberID] = principalSet[s.Members[0].MemberID]
-	}
+		// Go through the setters and apply them
+		for _, s := range f.setters {
+			members, ok := bindings[s.RoleID]
+			if !ok {
+				members = make(map[string]*models.HashicorpCloudResourcemanagerPolicyBindingMemberType, 4)
+				bindings[s.RoleID] = members
+			}
 
-	// Apply the policy
-	f.set(u.SetResourceIamPolicy(ctx, FromMap(ep.Etag, bindings)))
+			members[s.Members[0].MemberID] = principalSet[s.Members[0].MemberID]
+		}
 
+		// Apply the policy
+		ep, diags = u.SetResourceIamPolicy(ctx, FromMap(ep.Etag, bindings))
+		if diags.HasError() {
+			if customdiags.HasConflictError(diags) {
+				log.Printf("[DEBUG]: Operation failed due to conflicts. Operation will be restarted after %s\n", backoff)
```

Comment:
With log package you shouldn't need the newline character: https://go.dev/play/p/PCT9cNfOYGu